### PR TITLE
[Sol - Yul] Add support for built-in blockhash().

### PR DESCRIPTION
**Pull request by aarlt**
Part of #8343.

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -771,6 +771,15 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
+	case FunctionType::Kind::BlockHash:
+	{
+		solAssert(arguments.size() == 1, "");
+		IntegerType const* type = TypeProvider::uint(8);
+		IRVariable argument = convert(*arguments.front(), *type);
+		define(_functionCall) << "blockhash(" << argument.commaSeparatedList() << ")\n";
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]:
```suggestion
		define(_functionCall) << "blockhash(" << expressionAsType(*arguments.front()) << ")\n";
```

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -776,6 +776,13 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		define(_functionCall) << "gas()\n";
 		break;
 	}
+	case FunctionType::Kind::BlockHash:
+	{
+		solAssert(arguments.size() == 1, "");
+		IntegerType const* type = TypeProvider::uint(8);
+		define(_functionCall) << "blockhash(" << expressionAsType(*arguments.front(), *type) << ")\n";
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: What about
```suggestion
		define(_functionCall) << "blockhash(" << expressionAsType(*arguments.front(), *parameterTypes.front()) << ")\n";
```

Comment (aarlt) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: much better :)

**Comment by chriseth** [comment]
Can you maybe group this with selfdestruct and even gasleft?

**Comment by aarlt** [comment]
Yes, it is much better if similar things are grouped.

**Comment by aarlt** [comment]
Replaced by #8777

**Pull request closed without merging**